partner_contact_job_position/views/res_partner_view.xml

```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+<record model="ir.ui.view" id="view_partner_form_job_position">
+    <field name="name">Partner form with job position</field>
+    <field name="model">res.partner</field>
+    <field name="inherit_id" ref="base.view_partner_form"/>
+    <field name="arch" type="xml">
+        <field name="function" position="before">
+            <field name="job_position_id"
+                   attrs="{'invisible': [('is_company','=', True)]}"
+                   options='{"no_open": True}'/>
+        </field>
+        <xpath expr="//field[@name='child_ids']/form//field[@name='function']"
+               position="before">
+            <field name="is_company" invisible="True"/>
+            <field name="job_position_id"
+                   attrs="{'invisible': [('is_company','=', True)]}"
+                   options='{"no_open": True}'/>
+        </xpath>
+    </field>
+</record>
+
+<record model="ir.ui.view" id="view_res_partner_filter_job_position">
+    <field name="name">Partner search with job position</field>
+    <field name="model">res.partner</field>
+    <field name="inherit_id" ref="base.view_res_partner_filter"/>
+    <field name="arch" type="xml">
+        <field name="category_id" position="after">
+            <field name="job_position_id"/>
+        </field>
+        <filter name="salesperson" position="after">
+            <filter string="Job position"
+                    name="job_position_filter"
+                    domain="[('is_company','=', True)]"
```

Comment:
Question: why the domain on is_company? Maybe I am confused on the workings, but if we want to look for company contacts with a certain job position, will we not miss those contacts that are not companies?

Reply:
You are right, that is wrong, I' going to  fix it.